pkg/core/dao/dao.go

```diff
@@ -766,6 +786,21 @@ func (dao *Simple) DeleteBlock(h util.Uint256) error {
 	return nil
 }
 
+// DeleteHeader removes the header from dao. It's not atomic, so make sure you're
+// using private MemCached instance here.
+func (dao *Simple) DeleteHeader(h util.Uint256) error {
+	key := dao.makeExecutableKey(h)
+
+	// Check it's a block.
```

Comment:
Do we need this check?

Reply:
For the state reset functionality that uses this API - no (`(*Blockchain).resetStateInternal` does this check by himself), but I've added this check because it's an exported API and anyone can use it (possibly hoping that some check is performed).

Removed.